core/src/main/kotlin/org/eln2/sim/thermal/main.kt

```diff
@@ -0,0 +1,146 @@
+import kotlinx.coroutines.*
+import kotlinx.coroutines.sync.*
+import kotlin.math.*
+
+class ThermalInterface{
+
+    var mutex = Mutex()
+    var R = 1f //heat resistance at the interface
+    var heatTranfer = 0f //heat transfer during a simulation tick through this interface
+
+    var lastTickComputed = 0 //number of the last tick where the heat transfer was computed. Tick number switch between 0 and 1. This is here to prevent computing twice the heat trasnfered when not needed
+
+
+    fun computeHeatTransfer(dt : Float, tickNumber : Int){
+        var T0 = e0.getT(tickNumber)
+        var T1 = e1.getT(tickNumber)
+        if(tickNumber != lastTickComputed) {
+            heatTranfer = (T1 - T0) / R * dt
+            lastTickComputed = tickNumber
+        }
+    }
+
+    lateinit var e0 : ThermalElement
+    lateinit var e1 : ThermalElement
+}
+
+class SignedThermalInterface(thInterface : ThermalInterface, sgn : Float){
+    var mThermalInterface :ThermalInterface = thInterface
+    var sign : Float = sgn
+
+    fun computeHeatTransfer(dt : Float, tickNumber : Int):Float{
```

Comment:
One thing I noticed quite a lot through this section: there's a lot of `Float` here. I dislike that even the JVM makes you pick between Float and Double (it doesn't matter to Python, for example), but the space concerns are handled at the conversion boundaries, and there's no real performance benefit to using a float over a double in a managed language such as the JVM. For consistency's sake, our matrix math library seems to use doubles mostly, so...

Reply:
Okay I'll switch to double. I was wondering about that too.